force-app/main/default/classes/OpportunityHelper.cls

```diff
@@ -0,0 +1,240 @@
+    - Only updates the Opportunities that don't already have a primary contact.
+    */    
+
+public static void assignPrimaryContact(List<Opportunity> newOppList) {
+
+    // Collect IDs of opportunities needing a primary contact assignment
+    List<Opportunity> oppToUpdate = new List<Opportunity>();
+
+    // Find "VP Sales" contacts for the accounts associated with these opportunities
+    List<Contact> vpSalesContacts = [
+        SELECT Id, AccountId
+        FROM Contact
+        WHERE Title = 'VP Sales'
+        //WHERE Title LIKE '%VP Sales%' AND AccountId IN :oppMap.keySet()
+    ];
+
+    List<Contact> allContacts = [
+        SELECT Id, AccountId, Name, Title
+        FROM Contact
+
+        //WHERE Title LIKE '%VP Sales%' AND AccountId IN :oppMap.keySet()
+    ];
+
+    // Filter opportunities without a primary contact and not deleted
+    for (Opportunity opp : newOppList) {
+        if (opp.Primary_Contact__c == null && !opp.IsDeleted) {
+
+            // Clone the existing record to maintain original data and avoid modifying the original list directly
+            Opportunity updatedOpp = opp.clone(true, true, false); 
+            
+            updatedOpp.Primary_Contact__c = vpSalesContacts.get(0).Id;  // Set the Primary_Contact__c field
+
+            oppToUpdate.add(updatedOpp);  // Add the modified clone to the oppToUpdate list
+
+        }
+
+    }
+
+    // Optional: Perform the update if needed
+    try {
+        update oppToUpdate;
+        System.debug('Opportunities updated successfully: ' + oppToUpdate.size());
+    } catch (DmlException e) {
+        System.debug('Error updating Opportunities: ' + e.getMessage());
+    }
+
+}
+
+/*
+    setDefaultType:
+    - Assigns a Type for new Opportunities.
+    - Only updates the Opportunities that doesn't have an assigned Type.
+    */  
+    public static void setDefaultType(List<Opportunity> newOppList) {
+
+        // Set default Type for new Opportunities
+        for(Opportunity opp : newOppList) {
+            if (opp.Type == null){
+                    opp.Type = 'New Customer';
+            }        
+        }
+    }
+
+    /*
+    createNewTask:
+    - Create a new Task for newly inserted Opportunities
+    */  
+    public static void createNewTask(List<Opportunity> newOppList) {
+        // Check for Task read and create permissions
+        if (!Schema.sObjectType.Task.isAccessible()
+            && !Schema.sObjectType.Task.isCreateable()) {
+            throw new OtherException('Current user does not have permission to read and create Task records.');
+        }
+
+        List<Task> newTaskList = new List<Task>();
+
+        for (Opportunity opp : newOppList){
+            Task tsk = new Task();
+            tsk.Subject = 'Call Primary Contact';
+            tsk.WhatId = opp.Id;
+            tsk.WhoId = opp.Primary_Contact__c;
+            tsk.OwnerId = opp.OwnerId;
+            tsk.ActivityDate = Date.today().addDays(3);
```

Comment:
Try to use the date literals for practice and awareness
https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_select_dateformats.htm